src/App.js

```diff
@@ -80,32 +81,39 @@ const App = (props) => {
         } catch (e) {
             chrome.hideGlobalFilter();
         }
+
+        setHasSystemsReady(true);
     }
 
     useEffect(() => {
         initChrome();
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    if (!arePermissionsReady || !hasSystemsReady) {
+        return <PageLoading />;
+    }
+
+    if (hasSystemsReady && !hasSystems) {
+        return <ZeroState />;
+    }
+
```

Comment:
We should avoid multiple returns.

Reply:
What would be the reason? Is not this cleaner?

Reply:
The reason is that these if statements can get lost in code and usually developers look for the last return at the bottom.

It's also unexpected, as most components only have one return.

Reply:
I would prefer to have if statements in case, the return block has more than 3 condition factors. Because, in some apps, we have so complex conditionally returned component render blocks that it is hard to understand what condition renders what. Also, if we make a change to those conditions it is very easy to do a regression.

Reply:
If the function body would only be a handful of lines or even less, I would prefer multiple returns as well, but here it can be easy to scroll past and over look the loading and zero state and recognise them as a possible returned rendered state.